pages/stats.tsx

```diff
@@ -34,11 +49,29 @@ const Stats: PickleFinancePage = () => {
         <PlatformStats {...props} />
         <ChainStats {...props} />
         <JarStats {...props} />
+        <Loading {...props} />
       </div>
     </div>
   );
 };
 
+const Loading: FC<{ ready: readyState; page?: "platform" | "chain" | "jar" }> = ({
+  ready,
+  page,
+}) => {
+  if (page && ready[page] === false)
+    return (
+      <div className="flex justify-center items-center py-8 lg:py-32">
+        <div className="bg-background-light border border-foreground-alt-300 rounded-xl w-96 h-96">
+          <div className="flex justify-center mt-5">
+            <Image src="/animations/waiting.gif" alt="Loading..." width={250} height={250} />
+          </div>
+          <p className="text-xl text-center text-foreground-alt-200 mt-10">Loading Chart Data</p>
+        </div>
+      </div>
+    );
+  return null;
```

Comment:
It's more common to return early then return the rest, something like:

```suggestion
    if (ready[page]) return null;
    
    return (
      <div className="flex justify-center items-center py-8 lg:py-32">
        <div className="bg-background-light border border-foreground-alt-300 rounded-xl w-96 h-96">
          <div className="flex justify-center mt-5">
            <Image src="/animations/waiting.gif" alt="Loading..." width={250} height={250} />
          </div>
          <p className="text-xl text-center text-foreground-alt-200 mt-10">Loading Chart Data</p>
        </div>
      </div>
    );

```

Reply:
hmm I always thought the opposite made more sense in case something goes wrong or is not caught in the condition